### PR TITLE
Refactor utilities/imports.py

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Minor clean-up and sorted `_*_AVAILABLE` variables

## Before submitting
- [n/a] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

### CODE REVIEW

File: pytorch_lightning/utilities/imports.py [hunk 1]
```diff
@@ -30,33 +31,32 @@ def _module_available(module_path: str) -> bool:
     """
     # todo: find a better way than try / except
```
Comment (carmocca) [inline pytorch_lightning/utilities/imports.py:32]: @Borda I don't think there is a better way. The Python documentation clearly indicates that AttributeError/ModuleNotFoundError will be raised: https://docs.python.org/3/library/importlib.html#importlib.util.find_spec

File: pytorch_lightning/utilities/imports.py [hunk 2]
```diff
@@ -30,33 +31,32 @@ def _module_available(module_path: str) -> bool:
...
+
 _APEX_AVAILABLE = _module_available("apex.amp")
-_NATIVE_AMP_AVAILABLE = _module_available("torch.cuda.amp") and hasattr(torch.cuda.amp, "autocast")
-_OMEGACONF_AVAILABLE = _module_available("omegaconf")
+_BOLTS_AVAILABLE = _module_available('pl_bolts')
+_FAIRSCALE_AVAILABLE = not _IS_WINDOWS and _module_available('fairscale.nn.data_parallel')
+_FAIRSCALE_PIPE_AVAILABLE = (
+    _FAIRSCALE_AVAILABLE and LooseVersion(torch.__version__) >= LooseVersion("1.6.0")
+    and LooseVersion(pkg_resources.get_distribution('fairscale').version) <= LooseVersion("0.1.3")
+)
```
Comment (Borda) [inline pytorch_lightning/utilities/imports.py:51]:
```suggestion
_TORCH_GREATER_EQUAL_1_6 = LooseVersion(torch.__version__) >= LooseVersion("1.6.0")
_FAIRSCALE_PIPE_AVAILABLE = (
    _FAIRSCALE_AVAILABLE and _TORCH_GREATER_EQUAL_1_6
    and _get_version('fairscale') <= LooseVersion("0.1.3")
)
```

File: pytorch_lightning/utilities/imports.py [hunk 3]
```diff
@@ -30,33 +31,32 @@ def _module_available(module_path: str) -> bool:
...
-            if importlib.util.find_spec(module_path) is None:
-                return False
-        return True
+        return find_spec(module_path) is not None
     except AttributeError:
+        # Python 3.6
+        return False
+    except ModuleNotFoundError:
+        # Python 3.7+
         return False
```
Comment (Borda) [inline pytorch_lightning/utilities/imports.py:41]:
```suggestion


def _get_version(package: str) -> LooseVersion:
    return LooseVersion(pkg_resources.get_distribution(package).version)


```

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/utilities/imports.py [hunk 4]
```diff
@@ -30,33 +31,36 @@ def _module_available(module_path: str) -> bool:
...
+
+
+_IS_WINDOWS = platform.system() == "Windows"
+
 _APEX_AVAILABLE = _module_available("apex.amp")
-_NATIVE_AMP_AVAILABLE = _module_available("torch.cuda.amp") and hasattr(torch.cuda.amp, "autocast")
-_OMEGACONF_AVAILABLE = _module_available("omegaconf")
+_BOLTS_AVAILABLE = _module_available('pl_bolts')
+_FAIRSCALE_AVAILABLE = not _IS_WINDOWS and _module_available('fairscale.nn.data_parallel')
+_TORCH_GREATER_EQUAL_1_6 = LooseVersion(torch.__version__) >= LooseVersion("1.6.0")
```
Comment (SkafteNicki) [inline pytorch_lightning/utilities/imports.py:52]:
```suggestion
_TORCH_GREATER_EQUAL_1_6 = _get_version("torch") >= LooseVersion("1.6.0")
```

**Review by SkafteNicki — APPROVED** [review]

File: pytorch_lightning/utilities/imports.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/utilities/imports.py:32]: Given your :+1:, i'll remove the TODO again

```suggestion
```

File: pytorch_lightning/utilities/imports.py [hunk 5]
```diff
@@ -30,33 +31,36 @@ def _module_available(module_path: str) -> bool:
...
     except AttributeError:
+        # Python 3.6
+        return False
+    except ModuleNotFoundError:
+        # Python 3.7+
         return False
 
 
+def _get_version(package: str) -> LooseVersion:
+    return LooseVersion(pkg_resources.get_distribution(package).version)
```
Comment (carmocca) [inline pytorch_lightning/utilities/imports.py:43]: @Borda can you add a comment for why this is neccessary?

File: pytorch_lightning/utilities/imports.py [hunk 6]
```diff
@@ -12,51 +12,54 @@
...
+
+
+_IS_WINDOWS = platform.system() == "Windows"
+
 _APEX_AVAILABLE = _module_available("apex.amp")
-_NATIVE_AMP_AVAILABLE = _module_available("torch.cuda.amp") and hasattr(torch.cuda.amp, "autocast")
-_OMEGACONF_AVAILABLE = _module_available("omegaconf")
+_BOLTS_AVAILABLE = _module_available('pl_bolts')
+_FAIRSCALE_AVAILABLE = not _IS_WINDOWS and _module_available('fairscale.nn.data_parallel')
+_TORCH_GREATER_EQUAL_1_6 = _get_version("torch") >= LooseVersion("1.6.0")
```
Comment (carmocca) [inline pytorch_lightning/utilities/imports.py:51]:
```suggestion
_TORCH_GREATER_EQUAL_1_6 = _get_version("torch") >= LooseVersion("1.6.0")
```

File: pytorch_lightning/utilities/imports.py [hunk 7]
```diff
@@ -12,51 +12,54 @@
...
-_FAIRSCALE_PIPE_AVAILABLE = _FAIRSCALE_AVAILABLE and LooseVersion(
-    torch.__version__
-) >= LooseVersion("1.6.0") and LooseVersion(pkg_resources.get_distribution('fairscale').version
-                                            ) <= LooseVersion("0.1.3")
-_BOLTS_AVAILABLE = _module_available('pl_bolts')
+_NATIVE_AMP_AVAILABLE = _module_available("torch.cuda.amp") and hasattr(torch.cuda.amp, "autocast")
+_OMEGACONF_AVAILABLE = _module_available("omegaconf")
 _PYTORCH_PRUNE_AVAILABLE = _module_available('torch.nn.utils.prune')
+_RPC_AVAILABLE = not _IS_WINDOWS and _module_available('torch.distributed.rpc')
+_TORCHTEXT_AVAILABLE = _module_available("torchtext")
```
Comment (carmocca) [inline pytorch_lightning/utilities/imports.py:63]:
```suggestion
_RPC_AVAILABLE = not _IS_WINDOWS and _module_available('torch.distributed.rpc')
_TORCH_GREATER_EQUAL_1_6 = _get_version("torch") >= LooseVersion("1.6.0")
_TORCHTEXT_AVAILABLE = _module_available("torchtext")
```

File: pytorch_lightning/utilities/imports.py (hunk 5 above)
Comment (Borda) [reply pytorch_lightning/utilities/imports.py:43]: just wrapper and make the following notations simpler

File: pytorch_lightning/utilities/imports.py [hunk 8]
```diff
@@ -12,51 +12,54 @@
...
+    except ModuleNotFoundError:
+        # Python 3.7+
         return False
 
 
+def _get_version(package: str) -> LooseVersion:
+    return LooseVersion(pkg_resources.get_distribution(package).version)
+
+
+_IS_WINDOWS = platform.system() == "Windows"
```
Comment (carmocca) [inline pytorch_lightning/utilities/imports.py:46]:
```suggestion
_IS_WINDOWS = platform.system() == "Windows"
_TORCH_GREATER_EQUAL_1_6 = _get_version("torch") >= LooseVersion("1.6.0")
```

File: pytorch_lightning/utilities/imports.py [hunk 9]
```diff
@@ -12,51 +12,54 @@
...
-_FAIRSCALE_PIPE_AVAILABLE = _FAIRSCALE_AVAILABLE and LooseVersion(
-    torch.__version__
-) >= LooseVersion("1.6.0") and LooseVersion(pkg_resources.get_distribution('fairscale').version
-                                            ) <= LooseVersion("0.1.3")
-_BOLTS_AVAILABLE = _module_available('pl_bolts')
+_NATIVE_AMP_AVAILABLE = _module_available("torch.cuda.amp") and hasattr(torch.cuda.amp, "autocast")
+_OMEGACONF_AVAILABLE = _module_available("omegaconf")
 _PYTORCH_PRUNE_AVAILABLE = _module_available('torch.nn.utils.prune')
+_RPC_AVAILABLE = not _IS_WINDOWS and _module_available('torch.distributed.rpc')
+_TORCH_GREATER_EQUAL_1_6 = _get_version("torch") >= LooseVersion("1.6.0")
```
Comment (carmocca) [inline pytorch_lightning/utilities/imports.py:62]:
```suggestion
```

**Review by rohitgr7 — APPROVED** [review]
